### PR TITLE
Add atomic_and and atomic_or tests and wrappers for desul atomics

### PR DESCRIPTION
**Pull request by crtrott**
These were missing since atomic_and and aotmic_or were never used in Kokkos Core, or KokkosKernels

### CODE REVIEW

File: core/src/Kokkos_Atomics_Desul_Volatile_Wrapper.hpp [hunk 1]
```diff
@@ -126,6 +126,14 @@ void atomic_min(volatile T* const dest, desul::Impl::dont_deduce_this_parameter_
 template<class T> KOKKOS_INLINE_FUNCTION
 void atomic_max(volatile T* const dest, desul::Impl::dont_deduce_this_parameter_t<const T> val) { return desul::atomic_max (const_cast<T*>(dest), val, desul::MemoryOrderRelaxed(), desul::MemoryScopeDevice()); }
 
+// Desult doesn't have atomic_and yet so call fetch_and
```
Comment (masterleinad) [inline core/src/Kokkos_Atomics_Desul_Volatile_Wrapper.hpp:129]:
```suggestion
// FIXME Desul doesn't have atomic_and yet so call fetch_and
```

File: core/src/Kokkos_Atomics_Desul_Volatile_Wrapper.hpp [hunk 2]
```diff
@@ -126,6 +126,14 @@ void atomic_min(volatile T* const dest, desul::Impl::dont_deduce_this_parameter_
...
+template<class T> KOKKOS_INLINE_FUNCTION
+void atomic_and(volatile T* const dest, desul::Impl::dont_deduce_this_parameter_t<const T> val) { (void) desul::atomic_fetch_and (const_cast<T*>(dest), val, desul::MemoryOrderRelaxed(), desul::MemoryScopeDevice()); }
+
+// Desult doesn't have atomic_or yet so call fetch_or
```
Comment (masterleinad) [inline core/src/Kokkos_Atomics_Desul_Volatile_Wrapper.hpp:133]:
```suggestion
// FIXME Desul doesn't have atomic_or yet so call fetch_or
```

File: core/src/Kokkos_Atomics_Desul_Wrapper.hpp [hunk 3]
```diff
@@ -162,6 +162,14 @@ void atomic_min(T* const dest, desul::Impl::dont_deduce_this_parameter_t<const T
 template<class T> KOKKOS_INLINE_FUNCTION
 void atomic_max(T* const dest, desul::Impl::dont_deduce_this_parameter_t<const T> val) { return desul::atomic_max (dest, val, desul::MemoryOrderRelaxed(), desul::MemoryScopeDevice()); }
 
+// Desult doesn't have atomic_and yet so call fetch_and
```
Comment (masterleinad) [inline core/src/Kokkos_Atomics_Desul_Wrapper.hpp:165]:
```suggestion
// FIXME Desul doesn't have atomic_and yet so call fetch_and
```

File: core/src/Kokkos_Atomics_Desul_Wrapper.hpp [hunk 4]
```diff
@@ -162,6 +162,14 @@ void atomic_min(T* const dest, desul::Impl::dont_deduce_this_parameter_t<const T
...
+template<class T> KOKKOS_INLINE_FUNCTION
+void atomic_and(T* const dest, desul::Impl::dont_deduce_this_parameter_t<const T> val) { (void) desul::atomic_fetch_and (dest, val, desul::MemoryOrderRelaxed(), desul::MemoryScopeDevice()); }
+
+// Desult doesn't have atomic_or yet so call fetch_or
```
Comment (masterleinad) [inline core/src/Kokkos_Atomics_Desul_Wrapper.hpp:169]:
```suggestion
// FIXME Desul doesn't have atomic_or yet so call fetch_or
```

**Review by masterleinad — APPROVED** [review]
I would add some `FIXME`s so that we don't forget it.

File: core/src/Kokkos_Atomics_Desul_Volatile_Wrapper.hpp (hunk 1 above)
Comment (crtrott) [reply core/src/Kokkos_Atomics_Desul_Volatile_Wrapper.hpp:129]: done

File: core/src/Kokkos_Atomics_Desul_Volatile_Wrapper.hpp (hunk 2 above)
Comment (crtrott) [reply core/src/Kokkos_Atomics_Desul_Volatile_Wrapper.hpp:133]: done

File: core/src/Kokkos_Atomics_Desul_Wrapper.hpp (hunk 3 above)
Comment (crtrott) [reply core/src/Kokkos_Atomics_Desul_Wrapper.hpp:165]: done

File: core/src/Kokkos_Atomics_Desul_Wrapper.hpp (hunk 4 above)
Comment (crtrott) [reply core/src/Kokkos_Atomics_Desul_Wrapper.hpp:169]: done

**Review by dalg24 — APPROVED** [review]

**Pull request merged**